src/instrumenter/instrument.ts

```diff
@@ -280,62 +275,102 @@ function getDebugInfo(
     const factory = transCtx.factory;
     const instrCtx = transCtx.instrCtx;
 
+    assert(transCtx.container.vScope instanceof ContractDefinition, ``);
+
+    const events = resolveByName(transCtx.container.vScope, EventDefinition, "AssertionFailedData");
+    let evtDef = undefined;
+    if (events.length > 0) {
+        evtDef = events[0];
+    }
+
+    assert(
+        instrCtx.encodedLoggerArgs != null,
+        "Logger Args are not set during the debug events run"
+    );
+
     for (const annot of annotations) {
         const dbgIds = [...gatherDebugIds(annot.expression, transCtx.typeEnv)];
 
         if (dbgIds.length == 0) {
             res.push(undefined);
         } else {
-            // Next construct the parameters for the event
-            const evtParams = dbgIds.map((v) => {
+            const evtArgs = dbgIds.map((v) => transpile(v, transCtx));
+            const typeList: Array<[string, string]> = dbgIds.map((v) => {
                 const vType = transCtx.typeEnv.typeOf(v);
-                const type = transpileType(vType, transCtx.factory);
                 const typeString = vType instanceof PointerType ? vType.to.pp() : vType.pp();
+                return [v.name, typeString];
+            });
+            if (!instrCtx.debugEventsEncoding.has(annot.id)) {
+                instrCtx.debugEventsEncoding.set(annot.id, typeList);
+            }
 
-                return factory.makeVariableDeclaration(
+            const state = transCtx.refBinding(instrCtx.encodedLoggerArgs);
+            const stateAssignment = factory.makeExpressionStatement(
```

Comment:
You don't actually need a separate `encoded_logger_args` variable. You can pass the `abi.encodeWithSignature` `FunctionCall` directly into the arguments to the `Emit` statement (on line 368 instead of `state`).

This would allow you to also remove the `encodedLoggerArgs` from the `InstrumentationContext` class.